src/plot_api/plot_api.js

```diff
@@ -2093,7 +2093,7 @@ Plotly.update = function update(gd, traceUpdate, layoutUpdate, traces) {
     // fill in redraw sequence
     var seq = [];
 
-    if(restyleFlags.fullReplot && relayoutFlags.layoutReplot) {
+    if(restyleFlags.fullReplot || relayoutFlags.layoutReplot) {
```

Comment:
@etpinard am I missing something here? Seems like we would have noticed this before if it's a bug, but clicktoshow wouldn't work right without this change.